Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
 end
 
---- Set a given player's plusOne value to 0
+--- Check whether plus ones are enabled
 ---
----@param playerName string
----@return number
-function PlusOnes:setToZero(playerName)
-    GL:debug("PlusOnes:setToZero");
+---@return boolean
+function PlusOnes:enabled()
+    GL:debug("PlusOnes:enabled");
+
+    return GL.Settings:get("PlusOnes.enabled", false);
+end
+
+--- Check whether there are plus ones available
+---
+---@return boolean
+function PlusOnes:available()
+    GL:debug("PlusOnes:available");
+
+    --return GL:higherThanZero(DB:get("PlusOnes.MetaData.updatedAt", 0)); --not sure this check is even needed for plus ones
+    return true;
+end
+
+--- Draw either the importer or overview
+--- based on the current boosted roll data
+---
+---@return void
+function PlusOnes:draw()
+    GL:debug("PlusOnes:draw");
+
+    -- Show the plus ones overview instead
+    GL.Interface.PlusOnes.Overview:draw();
+end
+
+--- Checks and handles whisper commands if enabled.
+---
+---@param _ string
+---@param message string
+---@param sender string
+---@return void
+function PlusOnes:handleWhisperCommand(_, message, sender)
+    GL:debug("PlusOnes:handleWhisperCommand");
+
+    -- Only listen to the following messages
+    if (not GL:strStartsWith(message, "!plusone")
+        and not GL:strStartsWith(message, "!PLUSONE")
+        and not GL:strStartsWith(message, "!po")
+        and not GL:strStartsWith(message, "!PO")
+        and not GL:strStartsWith(message, "!+1")
+    ) then
+        return;
+    end
+
+    local args = GL:strSplit(message, " ");
+
+    -- See if name is given.
+    if (#args > 1) then
+        local name = self:normalizedName(args[2]);
+        local plusOne = self:getPlusOnes(name);
+        GL:sendChatMessage(
+            string.format("Player %s's %s total is %d", GL:capitalize(name), GL.Settings:get("PlusOnes.identifier", "PO"), plusOne),
+            "WHISPER", nil, sender
+        );
+        return;
+    end
+
+    local name = sender;
+    if (not GL.isEra) then
+        name = GL:stripRealm(name);
+    end
+
+    name = self:normalizedName(name);
+    local plusOne = self:getPlusOnes(name);
+    GL:sendChatMessage(
+        string.format("Your %s total is %d", GL.Settings:get("PlusOnes.identifier", "PO"), plusOne),
+        "WHISPER", nil, sender
+    );
+end
+
+--- Materialize the plus one data to make it more accessible during runtime
+---
+---@return void
+function PlusOnes:materializeData()
+    GL:debug("PlusOnes:materializeData");
+
+    local DetailsByPlayerName = {}; -- Details including aliases by player name
+
+    --- Create entries from Totals data
+    for name, plusOne in pairs(DB:get("PlusOnes.Totals", {})) do
+        name = GL:normalizedName(name or "");
+        plusOne = self:toPlusOne(plusOne or 0);
+
+        if (type(name) == "string"
+            and not GL:empty(name)
+        ) then
+            if (not DetailsByPlayerName[name]) then
```

Comment:
Can just add `and not DetailsByPlayerName[name]` as a third check in the original if-statement

Reply:
I'll make the fix.